src/main/java/io/connect/scylladb/ScyllaDbSinkConnector.java

```diff
@@ -104,4 +107,84 @@ public ConfigDef config() {
   public String version() {
     return VersionUtil.getVersion();
   }
+
+  @Override
+  public Config validate(Map<String, String> connectorConfigs){
+
+
+    //do the usual validation, field by field
+    Config result = super.validate(connectorConfigs);
+
+    ConfigValue userConfig = getConfigValue(result, ScyllaDbSinkConnectorConfig.USERNAME_CONFIG);
+    ConfigValue passwordConfig = getConfigValue(result, ScyllaDbSinkConnectorConfig.PASSWORD_CONFIG);
+    if((userConfig.value() == null) && (passwordConfig.value() != null)){
+      userConfig.addErrorMessage("Username not provided, even though password is.");
+    }
+    if((userConfig.value() != null) && (passwordConfig.value() == null)){
+      userConfig.addErrorMessage("Password not provided, even though username is.");
+    }
+
+    ConfigValue securityEnable = getConfigValue(result, ScyllaDbSinkConnectorConfig.SECURITY_ENABLE_CONFIG);
+    if(securityEnable.value().equals(true)){
+      if(userConfig.value() == null){
+        userConfig.addErrorMessage("Username is required with security enabled.");
+      }
+      if(passwordConfig.value() == null){
+        passwordConfig.addErrorMessage("Password is required with security enabled.");
+      }
+    }
+
+    boolean hasErrors = result.configValues().stream().anyMatch(c -> !(c.errorMessages().isEmpty()) );
+    if(!hasErrors)
+    {
+      //Assuming that this config gets parsed and validated the same way as `Config result`
+      //I don't see a simple way to instantiate ScyllaDbSinkConnectorConfig with Config type right now.
+      config = new ScyllaDbSinkConnectorConfig(connectorConfigs);
+      //make trial connection
+      ScyllaDbSessionFactory sessionFactory = new ScyllaDbSessionFactory();
+
+      try (ScyllaDbSession session = sessionFactory.newSession(config)) {
+        if (!session.isValid()) {
+          ConfigValue contactPoints = getConfigValue(result, ScyllaDbSinkConnectorConfig.CONTACT_POINTS_CONFIG);
+          contactPoints.addErrorMessage("Session is invalid.");
+        } else {
+          ConfigValue keyspaceCreate = getConfigValue(result, ScyllaDbSinkConnectorConfig.KEYSPACE_CREATE_ENABLED_CONFIG);
+          ConfigValue keyspaceConfig = getConfigValue(result, ScyllaDbSinkConnectorConfig.KEYSPACE_CONFIG);
+          if (config.keyspaceCreateEnabled) {
+            if (keyspaceConfig.value() == null) {
+              //this is possibly not needed, since keyspace should be always required non-null.
+              keyspaceConfig.addErrorMessage("Scylladb keyspace name is required when keyspace creation is enabled.");
+            }
+          } else {
+
+            // check if keyspace exists: (this may not be the best way to do it)
+            ResultSet resultSet = session.executeQuery("SELECT keyspace_name FROM system_schema.keyspaces WHERE keyspace_name = '" + config.keyspace + "';");
```

Comment:
This `executeQuery` should not fill the query using string concatenation, rather it should bind the values with CQL:

```
session.executeStatement(new SimpleStatement("SELECT keyspace_name FROM system_schema.keyspaces WHERE keyspace_name = ?", config.keyspace))
```

An even better way to check if the keyspace is present would be `cluster.getMetadata().getKeyspace("keyspace") != null` (see `ScyllaDbSessionImpl` class for `cluster`) - maybe new `keyspaceExists` method is needed.